src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -910,5 +898,49 @@ private bool IsFailOverable(RequestFailedException rfe)
                    rfe.Status == (int)HttpStatusCode.RequestTimeout ||
                    rfe.Status >= (int)HttpStatusCode.InternalServerError;
         }
+
+        private async Task<Dictionary<string, ConfigurationSetting>> MapConfigurationSettings(Dictionary<string, ConfigurationSetting> data)
+        {
+            Dictionary<string, ConfigurationSetting> mappedData = new Dictionary<string, ConfigurationSetting>(StringComparer.OrdinalIgnoreCase);
+
+            foreach (KeyValuePair<string, ConfigurationSetting> kvp in data)
+            {
+                ConfigurationSetting setting = kvp.Value;
+                foreach (Func<ConfigurationSetting, ValueTask<ConfigurationSetting>> func in _options.Mappers)
+                {
+                    setting = await func(setting).ConfigureAwait(false);
+                }
+                if (setting != null)
+                {
+                    mappedData[kvp.Key] = setting;
+                }
+            }
+
+            return mappedData;
+        }
+
+        private IEnumerable<ConfigurationSetting> GetCurrentKeyValueCollection(string key, string label, IEnumerable<ConfigurationSetting> existingSettings)
+        {
+            IEnumerable<ConfigurationSetting> currentKeyValues;
+
+            if (key.EndsWith("*"))
+            {
+                // Get current application settings starting with changeWatcher.Key, excluding the last * character
+                var keyPrefix = key.Substring(0, key.Length - 1);

```

Comment:
`var keyPrefix`

Do you mind to switch that to `string keyPrefix`. We usually prefer explicit type when the type itself is not used in the assignment.

E.g. we use the following

`var dict = new Dictionary<string, string>();

but not

`var foo = Bar();`